rllib/core/rl_trainer/trainer_runner.py

```diff
@@ -334,7 +334,11 @@ def set_weights(self, weights) -> None:
         if self.is_local:
             self._trainer.set_weights(weights)
         else:
-            self._worker_manager.foreach_actor(lambda w: w.set_weights(weights))
+            results_or_errors = self._worker_manager.foreach_actor(
```

Comment:
This is added so that if set_weights() throws an error we catch it. This was surfaced during this PR.